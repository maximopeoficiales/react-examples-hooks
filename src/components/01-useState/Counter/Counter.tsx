import { useState } from "react";

interface MyProps {}
const defaultProps: MyProps = {};
const Counter = (props: MyProps) => {
  props = { ...defaultProps, ...props };

  const [counter, setCounter] = useState({
    counter1: 10,
    counter2: 30,
    counter3: 22,
  });
  const { counter1, counter2, counter3 } = counter;
  return (
    <div className="row justify-content-center ">
      <div className="col-md-8">
        <h1>Counter component</h1>
        <p>Counter1: {counter1}</p>
        <p>Counter2: {counter2}</p>
        <p>Counter3: {counter3}</p>
        <button
          className="btn btn-primary"
          onClick={() => {
            setCounter({ ...counter, counter1: counter1 + 1 });
          }}
        >
          Aumentar +1
        </button>
      </div>
    </div>
  );
};

export default Counter;
