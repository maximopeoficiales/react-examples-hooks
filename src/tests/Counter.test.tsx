import { shallow } from "enzyme";
import React from "react";
import Counter from "../components/01-useState/Counter/Counter";

describe("<Counter />", () => {
  test("debe renderizarse correctamente", () => {
    const wrapper = shallow(<Counter />);
    expect(wrapper).toMatchSnapshot();
  });
});
