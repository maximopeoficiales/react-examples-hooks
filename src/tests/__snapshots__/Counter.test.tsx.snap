// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> debe renderizarse correctamente 1`] = `
<div
  className="row justify-content-center "
>
  <div
    className="col-md-8"
  >
    <h1>
      Counter component
    </h1>
    <p>
      Counter1: 
      10
    </p>
    <p>
      Counter2: 
      30
    </p>
    <p>
      Counter3: 
      22
    </p>
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Aumentar +1
    </button>
  </div>
</div>
`;
